File: backend/semgrep_rules/tests/javascript/kisa_sw_24_insufficient_key_length.js
const crypto = require("crypto");
function unsafe() {
  // ruleid: kisa.sw24.javascript.insufficient-rsa-key-length
  return crypto.generateKeyPairSync("rsa", { modulusLength: 1024 });
}
function safe() {
  // ok: kisa.sw24.javascript.insufficient-rsa-key-length
  return crypto.generateKeyPairSync("rsa", { modulusLength: 3072 });
}

File: backend/semgrep_rules/tests/python/kisa_sw_24_insufficient_key_length.py
from Crypto.PublicKey import RSA
def unsafe():
    # ruleid: kisa.sw24.python.insufficient-rsa-key-length
    return RSA.generate(1024)
def safe():
    # ok: kisa.sw24.python.insufficient-rsa-key-length
    return RSA.generate(3072)
